examples_tests/02.ComputeShader/main.cpp

```diff
@@ -27,11 +38,584 @@ void main()
+
+		commandBuffers[i]->begin(0);
+
+		// Todo(achal): The fact that this pipeline barrier is solely on a compute queue might
+		// affect the srcStageMask. More precisely, I think, for some reason, that
+		// VK_PIPELINE_STAGE_TRANSFER_BIT shouldn't be specified in compute queue
+		// but present queue (or transfer queue if theres one??)
+		commandBuffers[i]->pipelineBarrier(asset::EPSF_TRANSFER_BIT,
+			asset::EPSF_COMPUTE_SHADER_BIT, 0, 0u, nullptr, 0u, nullptr, 1u,
+			&undefToComputeTransitionBarrier);
+
+		commandBuffers[i]->bindComputePipeline(pipeline.get());
+
+		const video::IGPUDescriptorSet* tmp[] = { descriptorSets[i].get() };
+		commandBuffers[i]->bindDescriptorSets(asset::EPBP_COMPUTE, pipelineLayout.get(),
+			0u, 1u, tmp);
+
+		commandBuffers[i]->dispatch(WIN_W, WIN_H, 1);
+
+		commandBuffers[i]->pipelineBarrier(asset::EPSF_COMPUTE_SHADER_BIT, asset::EPSF_BOTTOM_OF_PIPE_BIT,
+			0, 0u, nullptr, 0u, nullptr, 1u, &computeToPresentTransitionBarrier);
+
+		commandBuffers[i]->end();
+	}
+
+	video::ISwapchain* rawPointerToSwapchain = swapchain.get();
+	
+	uint32_t currentFrameIndex = 0u;
+	while (!windowShouldClose_Global)
+	{
+		video::IGPUSemaphore* acquireSemaphore_frame = acquireSemaphores[currentFrameIndex].get();
+		video::IGPUSemaphore* releaseSemaphore_frame = releaseSemaphores[currentFrameIndex].get();
+		video::IGPUFence* fence_frame = frameFences[currentFrameIndex].get();
+
+		video::IGPUFence::E_STATUS retval = device->waitForFences(1u, &fence_frame, true, ~0ull);
+		assert(retval == video::IGPUFence::ES_SUCCESS);
+
+		uint32_t imageIndex;
+		swapchain->acquireNextImage(~0ull, acquireSemaphores[currentFrameIndex].get(), nullptr,
+			&imageIndex);
+
+		// Make sure you unsignal the fence before expecting vkQueueSubmit to signal it
+		// once it finishes its execution
+		device->resetFences(1u, &fence_frame);
+
+		// Todo(achal): Not really sure why are waiting at this pipeline stage for
+		// acquiring the image to render
+		asset::E_PIPELINE_STAGE_FLAGS waitDstStageFlags = asset::E_PIPELINE_STAGE_FLAGS::EPSF_COLOR_ATTACHMENT_OUTPUT_BIT;
+		
+		video::IGPUQueue::SSubmitInfo submitInfo = {};
+		submitInfo.waitSemaphoreCount = 1u;
+		submitInfo.pWaitSemaphores = &acquireSemaphore_frame;
+		submitInfo.pWaitDstStageMask = &waitDstStageFlags;
+		submitInfo.signalSemaphoreCount = 1u;
+		submitInfo.pSignalSemaphores = &releaseSemaphore_frame;
+		submitInfo.commandBufferCount = 1u;
+		submitInfo.commandBuffers = &commandBuffers[imageIndex].get();
+
+		computeQueue->submit(1u, &submitInfo, fence_frame);
+
+		video::IGPUQueue::SPresentInfo presentInfo;
+		presentInfo.waitSemaphoreCount = 1u;
+		presentInfo.waitSemaphores = &releaseSemaphore_frame;
+		presentInfo.swapchainCount = 1u;
+		presentInfo.swapchains = &rawPointerToSwapchain;
+		presentInfo.imgIndices = &imageIndex;
+		presentQueue->present(presentInfo);
+
+		currentFrameIndex = (currentFrameIndex + 1) % FRAMES_IN_FLIGHT;
+	}
+	device->waitIdle();
```

Comment:
most of this should be in commonAPI by now

Reply:
resolved